dallo/src/types.rs

```diff
@@ -5,10 +5,33 @@
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
 use rkyv::{
-    ser::serializers::AllocSerializer, ser::Serializer, Archive, Deserialize,
-    Infallible, Serialize,
+    ser::serializers::{
+        AllocSerializer, BufferScratch, BufferSerializer, CompositeSerializer,
+    },
+    ser::Serializer,
+    validation::validators::DefaultValidator,
+    Archive, Deserialize, Infallible, Serialize,
 };
 
+use bytecheck::CheckBytes;
+
+use crate::SCRATCH_BUF_BYTES;
+
+pub type StandardBufSerializer<'a> = CompositeSerializer<
+    BufferSerializer<&'a mut [u8]>,
+    BufferScratch<&'a mut [u8; SCRATCH_BUF_BYTES]>,
+>;
+
+pub trait StandardDeserialize<T>:
+    Deserialize<T, Infallible> + for<'a> CheckBytes<DefaultValidator<'a>>
+{
+}
+
+impl<T, U> StandardDeserialize<T> for U where
+    U: Deserialize<T, Infallible> + for<'a> CheckBytes<DefaultValidator<'a>>
+{
+}
```

Comment:
Consider not introducing this wrapper trait. The user will still have to just implement `Deserialize<_, Infallible>` and `CheckBytes<DefaultValidator<_>>`.